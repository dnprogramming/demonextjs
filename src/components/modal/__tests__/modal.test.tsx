import { render } from "@testing-library/react";
import Modal from "../modal";
import React from "react";
import GalleryContent from "@/content/gallery";

describe("Modal Component", () => {
    it ("renders modal with image", () => {
        const { getByAltText } = render(
            <Modal imageUrl={GalleryContent[0]} onClose={jest.fn()} />
        );
        const imageElement = getByAltText("Demo Next.JS Modal Image");
        expect(imageElement).toBeInTheDocument();
    });

    it("renders modal with close button", () => {
        const { getByText } = render(
            <Modal imageUrl={GalleryContent[0]} onClose={jest.fn()} />
        );
        const closeButton = getByText("×");
        expect(closeButton).toBeInTheDocument();
    });

    it("calls onClose when close button is clicked", () => {
        const onCloseMock = jest.fn();
        const { getByText } = render(
            <Modal imageUrl={GalleryContent[0]} onClose={onCloseMock} />
        );
        const closeButton = getByText("×");
        closeButton.click();
        expect(onCloseMock).toHaveBeenCalledTimes(1);
    });

    it("imageUrl is empty", () => {
        const onCloseMock = jest.fn();
        const { container } = render(
            <Modal imageUrl="" onClose={onCloseMock} />
        );
        expect(container).toBeInTheDocument();
    });

    it("renders modal component unchanged", () => {
        const { container } = render(<Modal imageUrl={GalleryContent[0]} onClose={jest.fn()} />);
        expect(container).toMatchSnapshot();
    });
});
