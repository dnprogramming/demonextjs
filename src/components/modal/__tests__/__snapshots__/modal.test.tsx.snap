// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Modal Component renders modal component unchanged 1`] = `
<div>
  <div
    class="modalOverlay"
  >
    <div
      class="modalContent"
    >
      <button
        class="closeButton"
      >
        Ã—
      </button>
      <img
        alt="Demo Next.JS Modal Image"
        class="modalImage"
        data-nimg="1"
        decoding="async"
        height="1080"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fhero_banner.png&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2Fhero_banner.png&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fhero_banner.png&w=3840&q=75 2x"
        style="color: transparent;"
        width="1920"
      />
    </div>
  </div>
</div>
`;
